common-ktx/src/main/java/co/nimblehq/common/extensions/ActivityExt.kt

```diff
@@ -0,0 +1,55 @@
+package co.nimblehq.common.extensions
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.view.WindowManager
+import android.view.inputmethod.InputMethodManager
+import androidx.annotation.ColorRes
+import androidx.core.content.ContextCompat
+
+/**
+ * Request to hide the soft input window from the Activity.
+ */
+fun Activity.hideSoftKeyboard() {
+    currentFocus?.let {
+        val inputMethodManager =
+            getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager
+        inputMethodManager?.hideSoftInputFromWindow(it.windowToken, 0)
+    }
+}
+
+/**
+ * Detect if the activity was launched from Recent Apps List.
+ *
+ * Reference: https://stackoverflow.com/q/4866149
+ */
+fun Activity.isActivityLaunchedFromHistory(): Boolean {
+    val flags = intent.flags
+    return flags and Intent.FLAG_ACTIVITY_LAUNCHED_FROM_HISTORY != 0
+}
+
+/**
+ * Restart the application by start the Activity in new task from recent history.
+ *
+ * @param clazz Activity class
+ */
+fun Activity.restartApp(clazz: Class<*>) {
+    val intent = Intent(this, clazz).apply {
+        addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+    }
+    startActivity(intent)
+    finish()
+    Runtime.getRuntime().exit(0)
+}
+
+/**
+ * Change the status bar color.
+ *
+ * @param colorResId Color resource id
+ */
+fun Activity.changeStatusBarColor(@ColorRes colorResId: Int) {
+    val window = this.window
```

Comment:
@hoangnguyen92dn Can we use  with`with(window)` ?

Reply:
@minhnimble @lydiasama I tried to apply `with` but AS shows redundant
![image](https://user-images.githubusercontent.com/6950766/142881926-a79e2270-b645-4087-9dae-17b5984721bc.png)